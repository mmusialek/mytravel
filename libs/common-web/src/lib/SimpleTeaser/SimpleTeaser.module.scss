@import "../styles/variables.scss";

.simpleTeaser {
  margin: 15px;

  &_image {
    max-height: 350px;
    max-width: 500px;
    position: relative;

    img {
      width: 100%;
      height: 100%;
    }

    &:hover>&_details {
      cursor: pointer;
      display: flex;
    }

    &_details {
      background-color: rgba($col-white, .7);
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      display: none;
      align-items: center;
      justify-content: center;

      &_link {
        background-color: rgba($col-white, .3);
        border: 2px solid #999;
        border-radius: 30px;
        font-size: 1.3em;
        font-weight: bold;
        padding: 15px 50px;
        text-decoration: none;
        color: black;

        &:hover {
          background-color: rgba($col-white, .6);
        }
      }
    }
  }

  &_info {
    font-size: .7em;
    color: $col-secondFont;
    padding-top: 15px;
  }

  &_title {
    font-size: 1.3em;
    font-weight: bold;
    color: black;
  }

  &_rating {
    font-size: .9em;
    font-weight: bold;
    padding: 15px 0;
    display: flex;
    align-items: center;

    &_value {
      padding: 0 0 5px 15px;
    }
  }

  &_price {
    display: flex;
    align-items: center;
    font-weight: bold;
    font-size: .9em;

    &__standard {
      color: black;
    }

    &__separator {
      width: 5px;
      height: 5px;
      background-color: $col-secondFont;
      border-radius: 50%;
      display: inline-block;
      margin: 0 10px;
    }

    &__promo {
      text-decoration: line-through;
      color: $col-secondFont;
    }

  }
}


// media section

@media only screen and (max-width: 600px) {
  .simpleTeaser {
    max-width: 350px;

    &:hover>&_details {
      cursor: unset;
      display: inherit;
    }

    &_image {
      &:hover>&_details {
        cursor: unset;
        display: none;
      }

      &_details {
        display: none;
      }
    }
  }
}


@media only screen and (min-width: 601px) and (max-width: 992px) {
  .simpleTeaser {
    max-width: 350px;

    &_image {
      &:hover>&_details {
        cursor: unset;
        display: none;
      }

      &_details {
        display: none;
      }
    }
  }
}
