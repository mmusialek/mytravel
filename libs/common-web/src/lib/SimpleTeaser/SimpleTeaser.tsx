import { IImageData } from "@mytravel/common-types";
import styles from "./SimpleTeaser.module.scss";


export interface ISimpleTeaserProps {
  image: IImageData;
  teaserInfo: string;
  title: string;
  rating: number;
  priceFrom: number;
  priceFromPromo?: number;
  currencySymbol: string;
}

export const SimpleTeaser = (props: ISimpleTeaserProps) => {


  const getStarRating = (rating: number) => {
    const res: React.ReactElement[] = [];
    const maxRating = 5;

    let tmp = rating;
    let ratingImage = "";
    for (let i = maxRating; i > 0; i--) {
      console.log(tmp);
      if (tmp >= 1) {
        ratingImage = "/assets/icons/star_full.svg";
      }

      if (tmp > 0 && tmp < 1) {
        ratingImage = "/assets/icons/star_half_full.svg";
      }

      if (tmp <= 0) {
        ratingImage = "/assets/icons/star_empty.svg";
      }

      res.push(<img src={ratingImage} alt="star" key={"star_rating_" + i} />)
      tmp -= 1;
    }

    return res;
  }

  return (<div className={styles.simpleTeaser}>
    {/* <div className={styles.simpleTeaser_image}>{props.image}</div> */}
    <div className={styles.simpleTeaser_image}>
      <img src={props.image.imagePath} alt={props.image.alt} />
      <div className={styles.simpleTeaser_image_details}>
        <a className={styles.simpleTeaser_image_details_link} href="#">Details</a>
      </div>
    </div>
    <div className={styles.simpleTeaser_info}>{props.teaserInfo}</div>
    <div className={styles.simpleTeaser_title}>{props.title}</div>
    <div className={styles.simpleTeaser_rating}>
      <span>{getStarRating(props.rating)}</span>
      <span className={styles.simpleTeaser_rating_value}>{props.rating}</span>
    </div>
    <div className={styles.simpleTeaser_price}>
      <span className={styles.simpleTeaser_price__standard}>From {props.currencySymbol}{props.priceFromPromo}</span>
      <span className={styles.simpleTeaser_price__separator}></span>
      <span className={styles.simpleTeaser_price__promo}>Price {props.currencySymbol}{props.priceFrom}</span>
    </div>
  </div >)
}
