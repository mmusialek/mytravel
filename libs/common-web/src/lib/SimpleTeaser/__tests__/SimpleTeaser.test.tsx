import renderer from "react-test-renderer";
import { ISimpleTeaserProps, SimpleTeaser } from "../SimpleTeaser";

describe("SimpleTeaser", () => {
  it("should render", () => {
    const props: ISimpleTeaserProps = {
      image: { imagePath: "image.jpg", alt: "mock image" },
      title: "mock title",
      teaserInfo: "some mock teaser info",
      rating: 3.4,
      currencySymbol: "$",
      priceFromPromo: 111,
      priceFrom: 222,
    };
    const comp = renderer.create(<SimpleTeaser {...props} />).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
