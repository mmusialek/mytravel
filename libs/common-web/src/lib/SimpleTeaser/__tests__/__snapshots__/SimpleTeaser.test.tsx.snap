// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTeaser should render 1`] = `
<div
  className="simpleTeaser"
>
  <div
    className="simpleTeaser_image"
  >
    <img
      alt="mock image"
      src="image.jpg"
    />
    <div
      className="simpleTeaser_image_details"
    >
      <a
        className="simpleTeaser_image_details_link"
        href="#"
      >
        Details
      </a>
    </div>
  </div>
  <div
    className="simpleTeaser_info"
  >
    some mock teaser info
  </div>
  <div
    className="simpleTeaser_title"
  >
    mock title
  </div>
  <div
    className="simpleTeaser_rating"
  >
    <span>
      <img
        alt="star"
        src="/assets/icons/star_full.svg"
      />
      <img
        alt="star"
        src="/assets/icons/star_full.svg"
      />
      <img
        alt="star"
        src="/assets/icons/star_full.svg"
      />
      <img
        alt="star"
        src="/assets/icons/star_half_full.svg"
      />
      <img
        alt="star"
        src="/assets/icons/star_empty.svg"
      />
    </span>
    <span
      className="simpleTeaser_rating_value"
    >
      3.4
    </span>
  </div>
  <div
    className="simpleTeaser_price"
  >
    <span
      className="simpleTeaser_price__standard"
    >
      From 
      $
      111
    </span>
    <span
      className="simpleTeaser_price__separator"
    />
    <span
      className="simpleTeaser_price__promo"
    >
      Price 
      $
      222
    </span>
  </div>
</div>
`;
